runtimes/pythonrt/pythonrt_test.go

```diff
@@ -129,6 +140,26 @@ func newCallbacks(svc *pySvc) *sdkservices.RunCallbacks {
 	return &cbs
 }
 
+func setupServer(l *zap.Logger) *http.Server {
+	ConfigureWorkerGRPCHandler(l)
+	server := &http.Server{
+		Addr:    "localhost:9980",
+		Handler: h2c.NewHandler(Server, &http2.Server{}),
+	}
+
+	if err := ConfigureLocalRunnerManager(l, LocalRunnerConfig{WorkerAddress: "localhost:9980"}); err != nil {
+		panic(err)
+	}
+
+	go func() {
+		if err := server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			panic(err)
```

Comment:
No panic please.

Reply:
done

Reply:
solved it with a channel and time.After, giving the server 2 seconds to start
should be more than enough, its just the worker PB, not full AK

let me know if that works for you or you have another option

```go
func setupServer(l *zap.Logger) (*http.Server, error) {
	mux := &http.ServeMux{}
	ConfigureWorkerGRPCHandler(l, mux)
	server := &http.Server{
		Addr:    "localhost:9980",
		Handler: h2c.NewHandler(mux, &http2.Server{}),
	}

	if err := ConfigureLocalRunnerManager(l, LocalRunnerConfig{WorkerAddress: "localhost:9980"}); err != nil {
		return nil, err
	}

	errChan := make(chan error)

	go func() {
		if err := server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
			errChan <- err
			return
		}
	}()

	select {
	case err := <-errChan:
		return nil, err
	case <-time.After(2 * time.Second): // give server time to start
		return server, nil
	}
}
```